client.go

```diff
@@ -252,6 +252,11 @@ func (client Client) Put(uri *url.URL, parameters url.Values) ([]byte, error) {
 	return client.nonIdempotentRequest("PUT", uri, parameters)
 }
 
+// PutFiles updates an object on the API, using an HTTP "PUT" request.
+func (client Client) PutFiles(uri *url.URL, parameters url.Values, files map[string][]byte) ([]byte, error) {
```

Comment:
Rather than a new method, I think we should mirror the signature of `Post` in `Put`.

`Post` is:
```go
func (client Client) Post(uri *url.URL, operation string, parameters url.Values, files map[string][]byte) ([]byte, error) {
	queryParams := url.Values{"op": {operation}}
	uri.RawQuery = queryParams.Encode()
	if files != nil {
		return client.nonIdempotentRequestFiles("POST", uri, parameters, files)
	}
	return client.nonIdempotentRequest("POST", uri, parameters)
}
```
So `Put`, currently:
```go
// Put updates an object on the API, using an HTTP "PUT" request.
func (client Client) Put(uri *url.URL, parameters url.Values) ([]byte, error) {
	return client.nonIdempotentRequest("PUT", uri, parameters)
}
```
Would become:
```Go
func (client Client) Put(uri *url.URL, parameters url.Values, files map[string][]byte) ([]byte, error) {
	if files != nil {
		return client.nonIdempotentRequestFiles("PUT", uri, parameters, files)
	}
	return client.nonIdempotentRequest("PUT", uri, parameters)
}
```
With 3 places to add the `nil` argument in calls.